app/main.py

```diff
@@ -1,6 +1,18 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> str:
+        keys = (args, tuple(sorted(kwargs.items())))
+        if keys in cache_dict:
+            print("Getting from cache")
+            return cache_dict[keys]
+        else:
+            print("Calculating new result")
+            cache_dict[keys] = func(*args, **kwargs)
+            return cache_dict[keys]
```

Comment:
```suggestion
        keys = (args, tuple(sorted(kwargs.items())))
        if keys in cache_dict:
            print("Getting from cache")
        else:
            print("Calculating new result")
            cache_dict[keys] = func(*args, **kwargs)
        return cache_dict[keys]
```

You can use `return` only once.